lib/optimizely.rb

```diff
@@ -74,7 +76,9 @@ def initialize( # rubocop:disable Metrics/ParameterLists
       config_manager = nil,
       notification_center = nil,
       event_processor = nil,
-      default_decide_options = []
+      default_decide_options = [],
+      odp_segments_cache = nil,
+      sdk_settings = {}
```

Comment:
> What about custom odp_segments_manager (segments api) and odp_event_manager (events api) injection? Not supported (for now)?

We can include them, wasn't sure what the plan was for that.

SdkSettings struct to include segments_manager, event_manager and segments_cache?

Reply:
In swift, this was added to include all future configs (not only for Odp) into the SdkSettings instead of adding too many params to constructor. I think each SDK may have own best solution to make it scalable.